src/pages/api/howManyProducts.ts

```diff
@@ -0,0 +1,44 @@
+import { NextApiRequest, NextApiResponse } from 'next';
+import { NUMBER_OF_PRODUCTS_COOKIE } from '../../constants/index';
+import { getDomain, setCookieOnResponseObject, redirectToError, redirectTo } from './apiUtils';
+import { isSessionValid } from './service/validator';
+import { InputCheck } from '../howManyProducts';
+
+export const isNumberOfProductsInvalid = (req: NextApiRequest): InputCheck => {
+    const { numberOfProductsInput = '' } = req.body;
+    const inputAsNumber = Number(numberOfProductsInput);
+    let error;
+    if (numberOfProductsInput === '' || Number.isNaN(inputAsNumber)) {
+        error = 'Enter a number';
+    } else if (!Number.isInteger(inputAsNumber) || Number(inputAsNumber) > 10 || Number(inputAsNumber) < 1) {
+        error = 'Enter a whole number between 1 and 10';
+    } else {
+        error = '';
+    }
+    const inputCheck = { numberOfProductsInput, error };
+    return inputCheck;
+};
+
+export default (req: NextApiRequest, res: NextApiResponse): void => {
+    try {
+        if (!isSessionValid(req, res)) {
+            throw new Error('Session is invalid.');
+        }
+
+        const userInputValidity = isNumberOfProductsInvalid(req);
```

Comment:
Maybe reverse logic so error first similar to this. So if error occurs it resolves that first. That way the else is not required

```
const userInputValidity = isNumberOfProductsInvalid(req);
        if (userInputValidity.error) {
            const numberOfProductsCookieValue = JSON.stringify(userInputValidity);
            setCookieOnResponseObject(getDomain(req), NUMBER_OF_PRODUCTS_COOKIE, numberOfProductsCookieValue, req, res);
            redirectTo(res, '/howManyProducts');

}
            const numberOfProductsCookieValue = JSON.stringify({
                numberOfProductsInput: userInputValidity.numberOfProductsInput,
            });
            setCookieOnResponseObject(getDomain(req), NUMBER_OF_PRODUCTS_COOKIE, numberOfProductsCookieValue, req, res);
            redirectTo(res, '/multipleProducts');
   
```

Reply:
Nice idea - I've changed this one around too